vm/src/compile.rs

```diff
@@ -226,29 +226,51 @@ impl Compiler {
             ast::Statement::Import { import_parts } => {
                 for ast::SingleImport {
                     module,
-                    symbol,
+                    symbols,
                     alias,
                 } in import_parts
                 {
-                    match symbol {
-                        Some(name) if name == "*" => {
-                            self.emit(Instruction::ImportStar {
-                                name: module.clone(),
-                            });
-                        }
-                        _ => {
+                    if let Some(alias) = alias {
+                        self.emit(Instruction::Import {
+                            name: module.clone(),
+                            symbols: vec![],
+                        });
+                        self.store_name(&alias);
+                    } else {
+                        if symbols.is_empty() {
                             self.emit(Instruction::Import {
                                 name: module.clone(),
-                                symbol: symbol.clone(),
+                                symbols: vec![],
                             });
-                            let name = match alias {
-                                Some(alias) => alias.clone(),
-                                None => match symbol {
-                                    Some(symbol) => symbol.clone(),
-                                    None => module.clone(),
-                                },
-                            };
-                            self.store_name(&name);
+                            self.store_name(&module.clone());
+                        } else {
+                            let mut import_star = false;
+                            let mut symbols_strings = vec![];
+                            let mut names = vec![];
```

Comment:
```rust
let import_star = symbols.iter().any(|ast::ImportSymbol { symbol, .. }| symbol == "*");
let (mut names, symbols_strings): (Vec<String>, Vec<String>) = symbols
    .into_iter().map(|ast::ImportSymbol { symbol, alias }| (alias.unwrap_or_else(|| symbol.to_string()), symbol))
    .unzip()
    .collect();
names.reverse();
```
I think that should work, and it has the benefit of using `length_hint()` from Vec so that it doesn't have to make a bunch of separate allocations.

Reply:
Nice. This is much simpler now :+1: